doc/user/migrating-existing-apis.md

```diff
@@ -0,0 +1,384 @@
+# Migrating Existing Kubernetes APIs
+
+Kubernetes APIs are assumed to evolve over time, hence the well-defined API [versioning scheme][k8s-versioning]. Upgrading your operator's APIs can be a non-trivial task, one that will involve changing quite a few source files and manifests. This document aims to identify the complexities of migrating an operator project's API using examples from existing operators.
+
+While examples in this guide follow particular types of API migrations, most of the documented migration steps can be generalized to all migration types. A thorough discussion of migration types for a particular project type (Go, Ansible, Helm) is found at the end of each project type's section.
+
+## Golang: Upgrading one Kind to a new Version from a Version with multiple Kinds
+
```

Comment:
Nit: The language is officially called [Go](https://golang.org/doc/faq#go_or_golang), which you use throughout this PR. Is there a reason you're using `Golang` here?

Reply:
Nope, changing.